Core/include/Acts/EventData/MultiComponentBoundTrackParameters.hpp

```diff
@@ -0,0 +1,240 @@
+      default;
+  ~MultiComponentBoundTrackParameters() = default;
+  MultiComponentBoundTrackParameters& operator=(
+      const MultiComponentBoundTrackParameters&) = default;
+  MultiComponentBoundTrackParameters& operator=(
+      MultiComponentBoundTrackParameters&&) = default;
+
+  /// Access the parameters
+  const auto& components() const { return m_components; }
+
+  /// Reference surface onto which the parameters are bound.
+  const Surface& referenceSurface() const { return *m_surface; }
+
+  /// Get a SingleBoundTrackParameters object for one component
+  std::pair<double, SingleParameters> operator[](std::size_t i) const {
+    return std::make_pair(
+        std::get<double>(m_components[i]),
+        SingleParameters(
+            m_surface, std::get<BoundVector>(m_components[i]),
+            std::get<std::optional<BoundSymMatrix>>(m_components[i])));
+  }
+
+  /// Parameters vector.
+  ParametersVector parameters() const {
+    return reduce([](const SingleParameters& p) { return p.parameters(); });
+  }
+
+  /// Optional covariance matrix.
+  std::optional<CovarianceMatrix> covariance() const {
+    const auto ret = reduce([](const SingleParameters& p) {
+      return p.covariance() ? *p.covariance() : CovarianceMatrix::Zero();
+    });
+
+    if (ret == CovarianceMatrix::Zero()) {
+      return std::nullopt;
+    } else {
+      return ret;
+    }
+  }
+
+  /// Space-time position four-vector.
+  ///
+  /// @param[in] geoCtx Geometry context for the local-to-global
+  /// transformation
+  Vector4 fourPosition(const GeometryContext& geoCtx) const {
+    return reduce(
+        [&](const SingleParameters& p) { return p.fourPosition(geoCtx); });
+  }
+
+  /// Spatial position three-vector.
+  ///
+  /// @param[in] geoCtx Geometry context for the local-to-global
+  /// transformation
+  Vector3 position(const GeometryContext& geoCtx) const {
+    return reduce(
+        [&](const SingleParameters& p) { return p.position(geoCtx); });
+  }
+
+  /// Time coordinate.
+  Scalar time() const {
+    return reduce([](const SingleParameters& p) { return p.time(); });
+  }
+
+  /// Unit direction three-vector, i.e. the normalized momentum
+  /// three-vector.
+  Vector3 unitDirection() const {
+    return reduce([](const SingleParameters& p) { return p.unitDirection(); })
+        .normalized();
+  }
+
+  /// Absolute momentum.
+  Scalar absoluteMomentum() const {
+    return reduce(
+        [](const SingleParameters& p) { return p.absoluteMomentum(); });
+  }
+
+  /// Transverse momentum.
+  Scalar transverseMomentum() const {
+    return reduce(
+        [](const SingleParameters& p) { return p.transverseMomentum(); });
+  }
+
+  /// Momentum three-vector.
+  Vector3 momentum() const {
+    return reduce([](const SingleParameters& p) { return p.momentum(); });
+  }
+
+  /// Particle electric charge.
+  Scalar charge() const {
+    return reduce([](const SingleParameters& p) { return p.charge(); });
+  }
+};
+
```

Comment:
A question here, for potential future optimisation ... in the Navigation, there's often the successive call `position()`, `direction()` ... would it make sense to implement a tuple call, such that the reduction has to be done only once ?

Reply:
This particular class (the `MultiComponentBoundTrackParameters<charge_t>`) should never interact with the navigation. All these accessors with the reductions are required by the `BoundTrackParametersConcept` in `TrackParametersConcept.hpp`.

However (as you also commented below), for the interaction between the `Stepper::State` and the `Navigator`, this could potentially save some computations.